### PR TITLE
Hide spotter chart when site has no spotter

### PR DESCRIPTION
**Pull request by ericboucher**
We noticed that when visiting a site with no spotter after visiting a site with a spotter, the spotter chart would sometimes stay on with data from the previous site.

This PR is adding an `hasSpotter` check before displaying spotter charts.

We could also look into making sure that `spotterData` get's removed properly when a new site is selected.

@tleraj let me know what you think

### CODE REVIEW

File: packages/website/src/store/Reefs/selectedReefSlice.ts [hunk 1]
```diff
@@ -99,6 +99,13 @@ const selectedReefSlice = createSlice({
       }
       return state;
     },
+    setReefSpotterData: (
```
Comment (SimplyBallistic) [inline packages/website/src/store/Reefs/selectedReefSlice.ts:102]: Could just leave as it is, but if the only action this is doing is clearing, we could rename to `clearReefSpotterData` so you only have a choice to fetch from api or clear.

**Review by SimplyBallistic — APPROVED** [review]
Overall looks good :+1:

**Review by tleraj — APPROVED** [review]
Looks good!!

File: packages/website/src/routes/ReefRoutes/Reef/index.tsx [hunk 2]
```diff
@@ -144,7 +144,7 @@ const Reef = ({ match, classes }: ReefProps) => {
       );
     } else {
       // Clear possible spotter data from previously selected reef
-      dispatch(setReefSpotterData(null));
+      dispatch(clearReefSpotterData(null));
```
Comment (ericboucher) [inline packages/website/src/routes/ReefRoutes/Reef/index.tsx:147]: @tleraj you could even go one step further and have `clearReefSpotterData()` instead and move the `set null` behavior to the slice.

**Pull request merged**